File: Assiut Sheets Python/sheet4/string_functions.py
n, q = [int(num) for num in input().split()]
s = input()

answers = []

for _ in range(q):
    query = input().split()
    if query[0] == "pop_back":
        s = s[:-1]

    elif query[0] == "front":
        answers.append(s[0])

    elif query[0] == "back":
        answers.append(s[-1])

    elif query[0] == "sort":
        l = int(query[1]) - 1
        r = int(query[2]) - 1
        s = s[:l] + \
            "".join(sorted(s[l:r+1])) + s[r+1:]

    elif query[0] == "reverse":
        l = int(query[1]) - 1
        r = int(query[2]) - 1
        s = s[:l] + \
            s[l:r+1][::-1] + s[r+1:]

    elif query[0] == "print":
        pos = int(query[1]) - 1
        answers.append(s[pos])

    elif query[0] == "substr":
        l = int(query[1]) - 1
        r = int(query[2]) - 1
        answers.append(s[l:r+1])

    elif query[0] == "push_back":
        s += query[1]

for answer in answers:
    print(answer)

File: assiut-sheets-2/sheet2/string-functions.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    int n, q;
    cin >> n >> q;
    string s;
    cin >> s;
    while (q--) {
        string x;
        cin >> x;
        if (x == "pop_back") s.pop_back();
        else if (x == "front") cout << s.front() << "\n";
        else if (x == "back") cout << s.back() << "\n";
        else if (x == "sort") {
            int l, r;
            cin >> l >> r;
            sort(s.begin()+l-1, s.begin()+r);
        }
        else if (x == "reverse") {
            int l, r;
            cin >> l >> r;
            reverse(s.begin()+l-1, s.begin()+r);
        }
        else if (x == "print") {
            int pos;
            cin >> pos;
            cout << s[pos-1] << "\n";
        }
        else if (x == "substr") {
            int l, r;
            cin >> l >> r;
            cout << s.substr(l-1, r-l+1) << "\n";
        }
        else if (x == "push_back") {
            char c;
            cin >> c;
            s.push_back(c);
        }
    }
    return 0;
}